okhttp/src/main/java/okhttp3/internal/cache/DiskLruCache.kt

```diff
@@ -386,664 +365,642 @@ public final class DiskLruCache implements Closeable, Flushable {
-    this.maxSize = maxSize;
-    if (initialized) {
-      executor.execute(cleanupRunnable);
+      entry = Entry(key)
+      lruEntries[key] = entry
     }
+    val editor = Editor(entry)
+    entry.currentEditor = editor
+    return editor
   }
 
   /**
    * Returns the number of bytes currently being used to store the values in this cache. This may be
    * greater than the max size if a background deletion is pending.
    */
-  public synchronized long size() throws IOException {
-    initialize();
-    return size;
+  @Synchronized @Throws(IOException::class)
+  fun size(): Long {
+    initialize()
+    return size
   }
 
-  synchronized void completeEdit(Editor editor, boolean success) throws IOException {
-    Entry entry = editor.entry;
-    if (entry.currentEditor != editor) {
-      throw new IllegalStateException();
-    }
+  @Synchronized @Throws(IOException::class)
+  internal fun completeEdit(editor: Editor, success: Boolean) {
+    val entry = editor.entry
+    check(entry.currentEditor == editor)
 
     // If this edit is creating the entry for the first time, every index must have a value.
     if (success && !entry.readable) {
-      for (int i = 0; i < valueCount; i++) {
-        if (!editor.written[i]) {
-          editor.abort();
-          throw new IllegalStateException("Newly created entry didn't create value for index " + i);
+      for (i in 0 until valueCount) {
+        if (!editor.written!![i]) {
+          editor.abort()
+          throw IllegalStateException("Newly created entry didn't create value for index $i")
         }
         if (!fileSystem.exists(entry.dirtyFiles[i])) {
-          editor.abort();
-          return;
+          editor.abort()
+          return
         }
       }
     }
 
-    for (int i = 0; i < valueCount; i++) {
-      File dirty = entry.dirtyFiles[i];
+    for (i in 0 until valueCount) {
+      val dirty = entry.dirtyFiles[i]
       if (success) {
         if (fileSystem.exists(dirty)) {
-          File clean = entry.cleanFiles[i];
-          fileSystem.rename(dirty, clean);
-          long oldLength = entry.lengths[i];
-          long newLength = fileSystem.size(clean);
-          entry.lengths[i] = newLength;
-          size = size - oldLength + newLength;
+          val clean = entry.cleanFiles[i]
+          fileSystem.rename(dirty, clean)
+          val oldLength = entry.lengths[i]
+          val newLength = fileSystem.size(clean)
+          entry.lengths[i] = newLength
+          size = size - oldLength + newLength
         }
       } else {
-        fileSystem.delete(dirty);
+        fileSystem.delete(dirty)
       }
     }
 
-    redundantOpCount++;
-    entry.currentEditor = null;
-    if (entry.readable | success) {
-      entry.readable = true;
-      journalWriter.writeUtf8(CLEAN).writeByte(' ');
-      journalWriter.writeUtf8(entry.key);
-      entry.writeLengths(journalWriter);
-      journalWriter.writeByte('\n');
+    val journalWriter = this.journalWriter!!
```

Comment:
This block is a good candidate for `this.journalWriter.apply { ... }`

Reply:
Dig.